components/core/src/clp_s/CommandLineArguments.cpp

```diff
@@ -0,0 +1,267 @@
+            "command-args",
+            po::value<std::vector<std::string>>()
+    );
+
+    po::positional_options_description general_positional_options_description;
+    general_positional_options_description.add("command", 1);
+    general_positional_options_description.add("command-args", -1);
+
+    po::options_description all_descriptions;
+    all_descriptions.add(general_options);
+    all_descriptions.add(general_positional_options);
+
+    try {
+        po::variables_map parsed_command_line_options;
+        po::parsed_options parsed = po::command_line_parser(argc, argv)
+                                            .options(all_descriptions)
+                                            .positional(general_positional_options_description)
+                                            .allow_unregistered()
+                                            .run();
+        po::store(parsed, parsed_command_line_options);
+        po::notify(parsed_command_line_options);
+
+        if (parsed_command_line_options.count("command") == 0) {
+            if (parsed_command_line_options.count("help") != 0) {
+                if (argc > 2) {
+                    SPDLOG_WARN("Ignoring all options besides --help.");
+                }
+
+                print_basic_usage();
+                std::cerr << "COMMAND is one of:" << std::endl;
+                std::cerr << "  c - compress" << std::endl;
+                std::cerr << "  x - decompress" << std::endl;
+                std::cerr << "  s - search" << std::endl;
+                std::cerr << std::endl;
+                std::cerr << "Try "
+                          << " c --help OR "
+                          << " x --help OR "
+                          << "q --help for command-specific details." << std::endl;
+
+                po::options_description visible_options;
+                visible_options.add(general_options);
+                std::cerr << visible_options << '\n';
+                return ParsingResult::InfoCommand;
+            }
+
+            throw std::invalid_argument("Command unspecified");
+        }
+
+        switch (command_input) {
+            case (char)Command::Compress:
+            case (char)Command::Extract:
+            case (char)Command::Search:
+                m_command = (Command)command_input;
+                break;
+            default:
+                throw std::invalid_argument(std::string("Unknown action '") + command_input + "'");
+        }
+
+        if (Command::Compress == m_command) {
+            po::options_description compression_positional_options;
+            compression_positional_options
+                    .add_options()("archive-dir", po::value<std::string>(&m_archive_dir)->value_name("DIR"), "output directory")(
+                            "input-paths",
+                            po::value<std::vector<std::string>>(&m_file_paths)->value_name("PATHS"),
+                            "input paths"
+                    );
```

Comment:
```suggestion
            // clang-format off
            compression_positional_options.add_options()(
                    "archives-dir",
                    po::value<std::string>(&m_archives_dir)->value_name("DIR"),
                    "output directory"
            )(
                    "input-paths",
                    po::value<std::vector<std::string>>(&m_file_paths)->value_name("PATHS"),
                    "input paths"
            );
            // clang-format on
```

Reply:
Note the change from `archive-dir` -> `archives-dir`. Lets call it archive**s** dir throughout.